Automattic/jetpack
# Add boost score bars to jetpack dashboard

Add Jetpack Boost module to Jetpack Dashboard

## Proposed changes:

This PR imports several packages into the Jetpack plugin ( The Boost Score Bar React component, the Boost client side API package, and the boost backend score package ) to add a new module to the dashboard to show users without boost their site score.

This modules:
 * Shows scores to users without boost
 * Allows them to install the plugin from the module
 * Does not show the score bars at all if the API errors out ( or shows previous scores if they exist )
 * Has several information icons that help give context to Boost to the user
 * Shows the user the last time their site score was tested from this module

### Other information:


## Jetpack product discussion

P2: p1HpG7-k9o-p2
Design: WOl1tRuvUmDecQ0Yj0Wh5G-fi-3013%3A4333

## Does this pull request change what data or activity we track or use?

Yes, Installing from the link below the "days since tested" text, and clicks on the information icons and the links in them. The added tracks are 
 * `boost_instant_tests_install`
 * `boost-grade-info-button`
 * `boost-conversion-loss-info-button`
 * `boost-conversion-loss-info-source`
 * `boost-critical-css-info-button`
 * `boost-critical-css-info-link`

## Testing instructions:

1. Checkout this branch locally or with the jetpack beta plugin
2. Without Boost installed or activated, go to the Jetpack dashboard: `/wp-admin/admin.php?page=jetpack#/dashboard`
3. Scroll to the Performance and Growth section. You should see the module loading like this
![image](https://github.com/Automattic/jetpack/assets/65001528/b4f2d6ea-cc2b-43dd-a971-691bcddc3627)
4. Once the plugins data is loaded, you should see the boost scores loading if this is your first time loading them from the API
![image](https://github.com/Automattic/jetpack/assets/65001528/b3b65a10-4de4-46ad-8bde-cef53901e273)
5. Once they are fully loaded, you should see something like this. Make sure the info icon buttons work correctly
![image](https://github.com/Automattic/jetpack/assets/65001528/85f8845b-62f9-486c-8643-784c3b3b1d06)
6. Click Install Boost from either the CTA or the `Activate Boost to run instant tests` link. You should see the CTA text or the Install text change into a verb such as "installing..." or "activating..."
![image](https://github.com/Automattic/jetpack/assets/65001528/298f9cce-97a6-4b9c-aad4-49da914ac089)
7. Once installed and activated without a paid Boost plan, you should see this, which lets users know about automated critical CSS with the paid plan. Make sure info icon button and CTA work correctly.
![image](https://github.com/Automattic/jetpack/assets/65001528/7d70ecc8-832b-4f19-8fa6-4c3905419e48)
8. Go to `/wp-admin/plugins.php` and disable boost (do not uninstall)
9. Go back to `/wp-admin/admin.php?page=jetpack#/dashboard` and you should now see something like this
![image](https://github.com/Automattic/jetpack/assets/65001528/529861e2-5d0a-4893-a642-4c4a6d5e3f84)
10. Click to Activate the plugin and once that is complete, you should see this again
![image](https://github.com/Automattic/jetpack/assets/65001528/5aca7744-e826-472f-aab9-dd9703d8bb8b)
11. Now go purchase a paid Boost plan via `https://wordpress.com/checkout/{your_site}/jetpack_boost_yearly`
12. Go back to  `/wp-admin/admin.php?page=jetpack#/dashboard` and you should see this
![image](https://github.com/Automattic/jetpack/assets/65001528/c6635f01-cb05-4685-aa74-afd43c1d5fc6)

Bonus: There are different messages that are used depending on what the scores are, if you are testing this locally, feel free to manually input the scores in `projects/plugins/jetpack/_inc/client/at-a-glance/boost/index.jsx` here to make sure the right ones pop up
